src/routes/userRoutes.js

```diff
@@ -0,0 +1,118 @@
+const express = require("express");
+const bodyParser = require("body-parser");
+const userRoutes = express.Router();
+const jwt = require("jsonwebtoken");
+const Users = require("../user.json");
+const validator = require("../helper/validator");
+const { addUser, updatePrefernce } = require("../helper/service");
+const bcrypt = require("bcryptjs");
+const verifyToken = require("../middleware/authJWT");
+userRoutes.use(bodyParser.urlencoded({ extended: false }));
+userRoutes.use(bodyParser.json());
+
+//To register a new user
+userRoutes.post("/register", async (req, res) => {
+  let userDetails = req.body;
+  let validatorResult = validator.userValidator(userDetails, Users);
+  if (validatorResult.status) {
+    try {
+      let userd = await addUser(userDetails);
+      res.status(200).send({ Status: "User registered Successfully" });
+    } catch (e) {
+      res.status(400).send(e);
+    }
+  } else {
+    res.status(400).send(validatorResult.message);
+  }
+});
+
+//To login a user
+userRoutes.post("/login", (req, res) => {
+  let userList = Users.Users;
+  let result = userList.filter((val) => val.email === req.body.email);
+  try {
+    if (result.length !== 0) {
```

Comment:
can follow the fail fast concept which will reduce these large if-else blocks as this becomes less readable once you are handling lots of failure cases. In this way, do all the validations first and throw errors as early as you can and then move on with the happy flow.